### ISSUE TITLE
TNS 3.0 Aot compilation failed

### ISSUE THREAD

**Issue by sis0k0** [NativeScript/nativescript-dev-webpack#143]
_From @naturalfreak on May 5, 2017 8:10_

tns version 3.0

> ERROR in ~/node_modules/nativescript-imagepicker/index.d.ts (1,29): Cannot find module 
> 'data/observable'.

package json

>  ```
> "dependencies": {
>     "@angular/animations": "^4.1.0",
>     "@angular/common": "^4.1.0",
>     "@angular/compiler": "^4.1.0",
>     "@angular/core": "^4.1.0",
>     "@angular/forms": "^4.1.0",
>     "@angular/http": "^4.1.0",
>     "@angular/platform-browser": "^4.1.0",
>     "@angular/platform-browser-dynamic": "^4.1.0",
>     "@angular/router": "^4.1.0",
>     "nativescript-angular": "^3.0.0",
>     "nativescript-camera": "^3.0.0",
>     "nativescript-fresco": "^3.0.2",
>     "nativescript-geolocation": "^3.0.0",
>     "nativescript-imagepicker": "^3.0.0",
>     "nativescript-iqkeyboardmanager": "^1.0.1",
>     "nativescript-telerik-ui-pro": "./packages/nativescript-ui-pro.tgz",
>     "nativescript-unit-test-runner": "^0.3.4",
>     "reflect-metadata": "^0.1.10",
>     "rxjs": "~5.3.0",
>     "tns-core-modules": "^3.0.0"
>   },
>   "devDependencies": {
>     "@angular/compiler-cli": "^4.1.0",
>     "@ngtools/webpack": "^1.3.1",
>     "babel-traverse": "^6.24.1",
>     "babel-types": "^6.24.1",
>     "babylon": "^6.17.0",
>     "codelyzer": "^3.0.0",
>     "copy-webpack-plugin": "~4.0.1",
>     "extract-text-webpack-plugin": "~2.1.0",
>     "filewalker": "^0.1.3",
>     "htmlparser2": "^3.9.2",
>     "jasmine-core": "^2.5.2",
>     "karma": "^1.6.0",
>     "karma-jasmine": "^1.1.0",
>     "karma-nativescript-launcher": "^0.4.0",
>     "lazy": "^1.0.11",
>     "nativescript-css-loader": "~0.26.1",
>     "nativescript-dev-android-snapshot": "^0.0.9",
>     "nativescript-dev-typescript": "^0.4.2",
>     "nativescript-dev-webpack": "^0.4.2",
>     "raw-loader": "~0.5.1",
>     "resolve-url-loader": "~2.0.2",
>     "tslint": "^5.1.0",
>     "typescript": "~2.3.2",
>     "webpack": "^2.5.0",
>     "webpack-sources": "~0.2.3",
>     "zone.js": "~0.8.8"
>   }
> ```
> tsconfing.json
>  ```
> {
>   "compilerOptions": {
>     "noImplicitAny": false,
>     "module": "commonjs",
>     "target": "es5",
>     "moduleResolution": "node",
>     "emitDecoratorMetadata": true,
>     "experimentalDecorators": true,
>     "noEmitHelpers": true,
>     "noEmitOnError": true,
>     "importHelpers": false,
>     "types": [],
>     "lib": [
>       "dom",
>       "es2015",
>       "es2015.iterable",
>       "es6"
>     ],
>     "baseUrl": ".",
>     "paths": {
>       "*": [
>         "./node_modules/tns-core-modules/*",
>         "./node_modules/*"
>       ]
>     }
>   },
>   "exclude": [
>     "node_modules",
>     "platforms",
>     "**/*.aot.ts"
>   ],
>   "angularCompilerOptions": {
>     "strictMetadataEmit": true,
>     "skipTemplateCodegen": true
>   },
>   "compileOnSave": false,
>   "buildOnSave": false
> }
>  ```
tsconfig.aot.json
```
{
    "extends": "./tsconfig",
    "compilerOptions": {
        "baseUrl": ".",
        "paths": {
            "ui/*": ["node_modules/tns-core-modules/ui/*"],
            "platform": ["node_modules/tns-core-modules/platform"],
            "image-source": ["node_modules/tns-core-modules/image-source"],
            "xml": ["node_modules/tns-core-modules/xml"],
            "xhr": ["node_modules/tns-core-modules/xhr"],
            "text": ["node_modules/tns-core-modules/text"],
            "data": ["node_modules/tns-core-modules/data"],
            "fetch": ["node_modules/tns-core-modules/fetch"],
            "trace": ["node_modules/tns-core-modules/trace"],
            "fps-meter": ["node_modules/tns-core-modules/fps-meter"],
            "color": ["node_modules/tns-core-modules/color"],
            "application-settings": ["node_modules/tns-core-modules/application-settings"],
            "http": ["node_modules/tns-core-modules/http"],
            "camera": ["node_modules/tns-core-modules/camera"],
            "console": ["node_modules/tns-core-modules/console"],
            "timer": ["node_modules/tns-core-modules/timer"],
            "utils/*": ["node_modules/tns-core-modules/utils/*"],
            "location": ["node_modules/tns-core-modules/location"],
            "file-system": ["node_modules/tns-core-modules/file-system"],
            "application": ["node_modules/tns-core-modules/application"],
            "image-asset": ["node_modules/tns-core-modules/image-asset"],
            "connectivity": ["node_modules/tns-core-modules/connectivity"],
            "globals": ["node_modules/tns-core-modules/globals"]
        }
    },
    "exclude": [
        "node_modules",
        "platforms"
    ],
    "angularCompilerOptions": {
        "skipMetadataEmit": true,
        "genDir": "./"
    }
}
```

_Copied from original issue: NativeScript/nativescript-imagepicker#87_

**Comment by sis0k0** [comment]
Hey, there! This issue was resolved in `nativescript-dev-webpack` 0.4.2. You'll have to delete your `tsconfig.aot.json` and the plugin will ship you the updated version.
You can also do it manually, by updating this row:

```
"data": ["node_modules/tns-core-modules/data"]
-->
"data/*": ["node_modules/tns-core-modules/data/*"]
```

**Comment by sis0k0** [comment]
Feel free to reopen if that doesn't resolve your problem.
